src/views/Viewer.vue

```diff
@@ -341,6 +344,14 @@ export default {
 				// get original mime
 				let mime = fileInfo.mime
 
+				// if we don't have a handler for this mime, abort
+				if (!(mime in this.components)) {
+					console.error('The following file could not be displayed', fileName, fileInfo)
+					showError(t('viewer', 'There is no plugin available to display this file type'))
+					this.close()
+					return
```

Comment:
Is there a way to check that upfront as an app calling openFile?

Reply:
Unless we use the filename to detect type, unfortunately no.
I started by doing this, but in the end nextcloud server have different mimetypes than some libraries you can find (and apps can register custom ones)

Unless we have a dedicated service that allow us to match the mimetype against server, I don't see how I can do it without :/